### PR TITLE
Revert os_unfair_lock usage

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

`OSAtomic*` and `OSSpinLock` are deprecated since macOS 10.12, iOS 10.0 and tvOS 10.0, but suggested replacement for OSSpinLock is missed at runtime on older versions, so we must use it until we support older versions.
~For modern platforms like Mac Catalyst where we can use the new API, define `OSSPINLOCK_USE_INLINED=1` to get inline implementations of the `OSSpinLock` interfaces in terms of the `<os/lock.h>` primitives.~

## Related PRs or issues

[AB#81586](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/81586)
#107

### CODE REVIEW

File: CrashReporter.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3613,6 +3613,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = "com.microsoft.${PRODUCT_NAME:rfc1034identifier}";
 				PRODUCT_NAME = "Tests iOS";
 				SDKROOT = iphoneos;
+				TARGETED_DEVICE_FAMILY = "1,2";
```
Comment (annakocheshkova) [inline CrashReporter.xcodeproj/project.pbxproj:3616]: Is this change related?

Comment (MatkovIvan) [reply CrashReporter.xcodeproj/project.pbxproj:3616]: This warning (see below) popped up when I fixed unit tests after first commit, so not directly but yes

`TARGETED_DEVICE_FAMILY value does not contain any device family values compatible with the Mac Catalyst platform. Please add the value '2' to the TARGETED_DEVICE_FAMILY build setting. (in target 'Tests iOS' from project 'CrashReporter')`
